### ISSUE TITLE
OrbitService aborted on start capture

### ISSUE THREAD

**Issue by paulrox** [google/orbit#4298]
HI,

I compiled orbit from the tag "current_release" (1.85) on Ubuntu 22.04 with GCC 11.2.0-19ubuntu1.

I had to solve some little issues to make it compile (such as the crashpad error) but I managed to make it work anyway.
The compilation ends with an error because 23 tests out of 3009 failed:
```
99% tests passed, 23 tests failed out of 3009
 
Total Test time (real) =  25.80 sec
 
The following tests did not run:
	240 - LinuxTracingIntegrationTest.SchedulingSlices (Skipped)
	241 - LinuxTracingIntegrationTest.FunctionCalls (Skipped)
	242 - LinuxTracingIntegrationTest.CallstackSamplesAndAddressInfos (Skipped)
	243 - LinuxTracingIntegrationTest.CallstackSamplesTogetherWithFunctionCalls (Skipped)
	244 - LinuxTracingIntegrationTest.CallstackSamplesWithFramePointers (Skipped)
	245 - LinuxTracingIntegrationTest.CallstackSamplesWithFramePointersTogetherWithFunctionCalls (Skipped)
	246 - LinuxTracingIntegrationTest.ThreadStateSlices (Skipped)
	247 - LinuxTracingIntegrationTest.ThreadNames (Skipped)
	248 - LinuxTracingIntegrationTest.ModuleUpdateOnDlopen (Skipped)
	249 - LinuxTracingIntegrationTest.GpuJobs (Skipped)
	251 - OrbitServiceIntegrationTest.FunctionCallsWithUserSpaceInstrumentation (Skipped)
	252 - OrbitServiceIntegrationTest.OrbitApi (Skipped)
	253 - OrbitServiceIntegrationTest.MemoryTracing (Skipped)
	445 - ServiceUtils.CategoriesTracepoints (Skipped)
	446 - ServiceUtils.NamesTracepoints (Skipped)
	462 - InstrumentProcessTest.FailToInstrumentAlreadyAttached (Skipped)
	466 - InstrumentProcessTest.GetErrorMessage (Skipped)
	490 - InstrumentFunctionTest.TooShort (Skipped)
	497 - InstrumentFunctionTest.Loop (Skipped)
	504 - InstrumentFunctionTest.UnconditionalJump8BitOffsetBackToBeginning (Skipped)
	505 - InstrumentFunctionTest.UnconditionalJump32BitOffsetBackToBeginning (Skipped)
	506 - InstrumentFunctionTest.ConditionalJump8BitOffsetBackToBeginning (Skipped)
	507 - InstrumentFunctionTest.ConditionalJump32BitOffsetBackToBeginning (Skipped)
	513 - UnwinderDeathTest.set_dex_files_error (Skipped)
	1605 - UnwindTest.remote_through_signal (Skipped)
	1606 - UnwindTest.remote_through_signal_sa_siginfo (Skipped)
	1607 - UnwindTest.remote_through_signal_with_invalid_func (Skipped)
	1608 - UnwindTest.remote_through_signal_sa_siginfo_with_invalid_func (Skipped)
 
The following tests FAILED:
	250 - OrbitServiceIntegrationTest.CaptureSmoke (Child aborted)
	455 - ExecuteInProcessTest.ExecuteInProcess (Child aborted)
	456 - ExecuteInProcessTest.ExecuteInProcessWithMicrosoftCallingConvention (Child aborted)
	459 - InjectLibraryInTraceeTest.OpenUseAndCloseLibraryInUserCode (Child aborted)
	460 - InjectLibraryInTraceeTest.OpenUseAndCloseLibraryInSyscall (Child aborted)
	465 - InstrumentProcessTest.Instrument (Child aborted)
	467 - InstrumentProcessTest.ExitPayloadDoesNotUseX87Fpu (Child aborted)
	488 - InstrumentFunctionTest.DoSomething (Child aborted)
	489 - InstrumentFunctionTest.CheckStackAlignedTo16Bytes (Child aborted)
	491 - InstrumentFunctionTest.LongEnough (Child aborted)
	492 - InstrumentFunctionTest.RipRelativeAddressing (Child aborted)
	493 - InstrumentFunctionTest.UnconditionalJump8BitOffset (Child aborted)
	494 - InstrumentFunctionTest.UnconditionalJump32BitOffset (Child aborted)
	495 - InstrumentFunctionTest.ConditionalJump8BitOffset (Child aborted)
	496 - InstrumentFunctionTest.ConditionalJump32BitOffset (Child aborted)
	498 - InstrumentFunctionTest.CheckIntParameters (Child aborted)
	499 - InstrumentFunctionTest.CheckFloatParameters (Child aborted)
	500 - InstrumentFunctionTest.CheckM256iParameters (Child aborted)
	501 - InstrumentFunctionTest.CheckIntParametersMsAbi (Child aborted)
	502 - InstrumentFunctionTest.CheckFloatParametersMsAbi (Child aborted)
	503 - InstrumentFunctionTest.CheckNoX87UnderflowInReturnTrampoline (Child aborted)
	508 - InstrumentFunctionTest.LongConditionalJump32BitOffsetBackToBeginning (Child aborted)
	509 - InstrumentFunctionTest.UnableToDisassembleBadInstruction (Child aborted)
Errors while running CTest
gmake: *** [Makefile:84: test] Error 8
ERROR: conanfile.py (OrbitProfiler/None): Error in build() method, line 167
	cmake.test(output_on_failure=True)
	ConanException: Error 2 while executing cmake --build '/home/***/orbit/build_default_relwithdebinfo' '--target' 'test' '--' '-j6'
```

Anyway, I'm able to start the OrbitService and the Orbit client apparently without any issue but when i start the capture on my application it gets a segmentation fault and also the OrbitService closes printing those lines:

```
[2022-10-20T14:21:41.751002] [UserSpaceInstrumentation/InstrumentProcess.cpp:372] Starting to instrument process with pid 25460
[2022-10-20T14:21:41.765875] [UserSpaceInstrumentation/InstrumentProcess.cpp:393] Injecting library "/home/***/orbit/build_default_relwithdebinfo/lib/liborbituserspaceinstrumentation.so" into process 25460
[2022-10-20T14:21:41.776752] [UserSpaceInstrumentation/ExecuteMachineCode.cpp:85] Fatal: Failed to wait for sigtrap after PTRACE_CONT. Expected pid: 25460 Pid returned from waitpid: 25460 status: 2943, WIFSTOPPED: 1, WSTOPSIG: 11
Aborted (core dumped)
```

I've tried the same version of Orbit on an Ubuntu 22.04 WSL2 distribution with the same GCC version and despite having the same compilation error due to failed tests, the capture works as expected.

Any suggestions?

**Comment by florian-kuebler** [comment]
Thanks for reporting this.

Assigning @danielfenner, as this looks like some issue in the user-space dynamic instrumentation. What you can do for now is, switch to "Kernel (Uprobes)" mode for dynamic instrumentation. This is slower, but considered to be more reliable.

For the capture options, you can simply click <img src="https://github.com/google/orbit/blob/main/icons/outline_settings_white_48dp.png?raw=true" width="18" height="18" alt="Settings icon" />.

**Comment by florian-kuebler** [comment]
Also the failing tests might be related to the issue and might help us to tackle this down.

However, as you already observed, when the tests run all builds are already done. So it is safe to start Orbit/OrbitService at that point.

**Comment by florian-kuebler** [comment]
This should have been also fixed by #4304
